index.js

```diff
@@ -138,47 +175,65 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
+let place = 1;
 
+function takeATicketNumber(queue) {
+  queue.push(place);
+  return `Welcome. You are ticket number ${place++}`;
+  //that you can increment the variable INSIDE the string iterpolation is v cool - Ben
+};
 
-// *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+//*** Uncomment the lines below to test
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
+function ticketNumberGeneratorFunc() {
+  let count = 1; 
+  //this works, but I am a little confused as to why the count isn't
+  //resetting to 1 every time takeATicketNumberFunc is called
+  //is it because takeATicketNumberFunc is just the RETURN value of the GeneratorFunc (the nested function)?
```

Comment:
Exactly this!
This is closure at work.
Inside the scope of the `takeATicketNumberFunc, the `count` variable is accessible, which means you can read and both write that variable.